### PR TITLE
Various enhancements

### PR DESCRIPTION
**Pull request by msoukhomlinov**
Added support for:
- Documenting HA configuration
- FortiGuard (under FortiCare) subscription status
- Split out Interfaces by interface type (inc including relevant info for each type)

Cosmetic changes
- Added additional fields for use in style/coverpage
- Added removal of blank columns from system and firewall tables

### CODE REVIEW

File: Src/Private/Get-AbrFgtSystem.ps1 [hunk 1]
```diff
@@ -96,7 +96,7 @@ function Get-AbrFgtSystem {
             }
 
             if ($info -and $settings -and $InfoLevel.System -ge 1) {
-                Section -Style Heading3 'GUI Settings' {
+                Section -Style Heading3 'Feature GUI visibility' {
```
Comment (alagoutte) [inline Src/Private/Get-AbrFgtSystem.ps1:99]: change also TableParams name ?

File: AsBuiltReport.Fortinet.FortiGate.Style.ps1 [hunk 2]
```diff
@@ -49,11 +49,11 @@ TableStyle -Id 'Borderless' -HeaderStyle Normal -RowStyle Normal -BorderWidth 0
 # Header & Footer
 if ($ReportConfig.Report.ShowHeaderFooter) {
     Header -Default {
-        Paragraph -Style Header "$($ReportConfig.Report.Name) - v$($ReportConfig.Report.Version)"
```
Comment (alagoutte) [inline AsBuiltReport.Fortinet.FortiGate.Style.ps1:52]: @tpcarman this change is ok ?

File: Src/Private/Get-AbrFgtSystem.ps1 [hunk 3]
```diff
@@ -319,11 +481,89 @@ function Get-AbrFgtSystem {
                 }
             }
 
+            # Fetch HA Configuration
+            $haConfig = Get-FGTSystemHA
```
Comment (alagoutte) [inline Src/Private/Get-AbrFgtSystem.ps1:485]: make a specific chapiter (and file) for HA ?

File: Src/Private/Get-AbrFgtSystem.ps1 [hunk 4]
```diff
@@ -319,11 +481,89 @@ function Get-AbrFgtSystem {
...
+            if( $haConfig.mode -ne 'standalone' -and $infoLevel.System -ge 1) {
+                Section -Style Heading3 'High Availability' {
+                    Paragraph "The following section details HA settings."
+                    BlankLine
+                
+                    Section -Style Heading4 'HA Configuration' {
+                        $OutObj = @()
+    
+                        $OutObj = [pscustomobject]@{
+                            "Group Name / ID / Mode" = $haConfig.'group-name'+' / '+$haConfig.'group-id'+' / '+$haConfig.mode
```
Comment (alagoutte) [inline Src/Private/Get-AbrFgtSystem.ps1:498]: one line by setting ? (like other)

File: Src/Private/Get-AbrFgtSystem.ps1 [hunk 5]
```diff
@@ -319,11 +481,89 @@ function Get-AbrFgtSystem {
...
+                            "HB Device" = $haConfig.'hbdev'
```
Comment (alagoutte) [inline Src/Private/Get-AbrFgtSystem.ps1:499]: can you align (indent) '=' ? (see visual code configuration for auto formatting)

**Comment by alagoutte** [comment]
Hi @msoukhomlinov 

Look great, i need to make some real try for look change !

May be better a PR by change i think !

File: AsBuiltReport.Fortinet.FortiGate.Style.ps1 (hunk 2 above)
Comment (tpcarman) [reply AsBuiltReport.Fortinet.FortiGate.Style.ps1:52]: The report style script is now redundant as it it replaced by the default style script in AsBuiltReport.Core v1.4.0 module, so users would not see this change unless they specified this style script.

File: AsBuiltReport.Fortinet.FortiGate.json [hunk 6]
```diff
@@ -1,6 +1,9 @@
 {
     "Report": {
         "Name": "Fortinet FortiGate As Built Report",
+        "Assets": "Your Firewall HostNames",
+        "ProjectRef": "Your Project Reference name and number",
+        "ClientName": "Your Client's Name",
```
Comment (tpcarman) [inline AsBuiltReport.Fortinet.FortiGate.json:6]: Client/Customer name is already defined in the AsBuiltReport Core configuration JSON file.

File: Src/Private/Get-AbrFgtFirewall.ps1 [hunk 7]
```diff
@@ -96,6 +96,29 @@ function Get-AbrFgtFirewall {
                         $TableParams['Caption'] = "- $($TableParams.Name)"
                     }
 
+                    # Identify and remove empty columns
```
[the remark this answers is not in the document]
Comment (msoukhomlinov) [reply Src/Private/Get-AbrFgtFirewall.ps1:99]: For example columns have alias/description, and if all rows have this column empty, then there's no point including it in the table.

[the remark this answers is not in the document]
Comment (alagoutte) [reply Src/Private/Get-AbrFgtFirewall.ps1:99]: Yes, may be add a settings for this 
No info/cofnigruation are an info :)

File: Src/Private/Get-AbrFgtFortiCare.ps1 [hunk 8]
```diff
@@ -57,6 +168,34 @@ function Get-AbrFgtForticare {
 
                 $OutObj | Table @TableParams
 
+
+                Paragraph "The following table details FortiGuard subscriptions and services on FortiGate."
+                BlankLine
+                
+                $OutObj = @()
+                foreach ($lincese in $licenseSummary) {
```
Comment (alagoutte) [inline Src/Private/Get-AbrFgtFortiCare.ps1:176]: typo $lincese (and after also)

File: Src/Private/Get-AbrFgtSystem.ps1 (hunk 1 above)
Comment (alagoutte) [reply Src/Private/Get-AbrFgtSystem.ps1:99]: and only to make also modification line 130 (TableParams = @{
                        Name         = "Settings"

File: AsBuiltReport.Fortinet.FortiGate.json (hunk 6 above)
Comment (msoukhomlinov) [reply AsBuiltReport.Fortinet.FortiGate.json:6]: @tpcarman, there's a usecase for having two sets of company details, one for service provider and other for client. What do you think?

Comment (tpcarman) [reply AsBuiltReport.Fortinet.FortiGate.json:6]: This sounds like a specific use case, not a generic one which would be needed by all users? If you want to insert specific details into your reports you should be creating a custom style script.

File: AsBuiltReport.Fortinet.FortiGate.Style.ps1 [hunk 9]
```diff
@@ -49,11 +49,11 @@ TableStyle -Id 'Borderless' -HeaderStyle Normal -RowStyle Normal -BorderWidth 0
...
+        Paragraph -Style Header "$($ReportConfig.Report.ClientName)     |     $($ReportConfig.Report.Name)     |     $($ReportConfig.Report.ProjectRef)"
```
[the remark this answers is not in the document]
Comment (tpcarman) [reply AsBuiltReport.Fortinet.FortiGate.Style.ps1:52]: Changing the header can be achieved by using a custom style script to tailor to your specific needs.

**Comment by alagoutte** [comment]
@msoukhomlinov can you split change on multiple PR ? it will be more easy to review and merge ?

**Comment by alagoutte** [comment]
@msoukhomlinov i merge majority of change (expect for split interface stuf and remove unused column)
i will look after to add split interface

**Pull request closed without merging**